nncf/quantization/algorithms/min_max/torch_backend.py

```diff
@@ -299,14 +298,13 @@ def _create_quantizer_insertion_command(
         target_point: PTTargetPoint,
         quantizer_config: QuantizerConfig,
         parameters: FakeQuantizeParameters,
-    ) -> PTInsertionCommand:
+    ) -> PTQuantizerInsertionCommand:
         _, scale_shape, _ = PTMinMaxAlgoBackend._get_input_scale_shape(nncf_graph, target_point, quantizer_config)
 
         quantizer = PTMinMaxAlgoBackend._create_quantizer(
             quantizer_config, scale_shape, parameters, target_point.target_type
         )
-
-        return PTInsertionCommand(target_point, quantizer, TransformationPriority.QUANTIZATION_PRIORITY)
+        return PTQuantizerInsertionCommand(target_point, quantizer)
 
```

Comment:
What's wrong with the regular `PTInsertionCommand` that it had to be changed? Even if you need to bring the generated commands to a specific interface, I would expect that `PTQuantizerInsertionCommand` would ultimately get turned into the old `PTInsertionCommand`. Otherwise you are breaking the priority functionality of insertions into the PT models.

Reply:
Our goal is to move quantizer insertion logic into model transformer as quantizer insertion requires quite a lot of additional steps (in some specific cases we need to insert quantizer into the NNCFNetwork, register its address in the External quantizer hook and put this hook into the NNCFNetwork as well).  Other backends don't need all this, so `PTQuantizerInsertionCommand` is aligning torch in following way: to insert a quantizer you need to 
* Create a quantizer instance
* Create insertion command
* Call model transformer with the command

Reply:
About priority functionality: we are planning to completely migrate to PTQ interface as PTQ has more algorithms for initialization, aligned with other backends and models after torch PTQ are trainable.

Reply:
> What's wrong with the regular PTInsertionCommand that it had to be changed?

Additional steps before call transformation for `PTInsertionCommand` brakes logic of work with common algorithms.
From `ModelTransformer` expected that all logic about insertion FQ will be done inside, without call another backend-specific functions. 

> Otherwise you are breaking the priority functionality of insertions into the PT models.

Please provide example of incorrect work of PT PTQ.

Reply:
> Other backends don't need all this

Which doesn't mean that you have to break existing functionality in torch in order to do this.

> About priority functionality: we are planning to completely migrate to PTQ interface as PTQ has more algorithms for initialization, aligned with other backends and models after torch PTQ are trainable.

Just what are you calling a PTQ interface? Whatever a PTQ interface is, it must follow and execute existing code that inserts quantizers in PT. If you need to issue `QuantizerInsertionCommand`s instead of `InsertionCommand`s - do it, but ultimately this must be translated into a regular `PTInsertionCommand` with a quantization priority.

Reply:
> Additional steps before call transformation for `PTInsertionCommand` brakes logic of work with common algorithms. From `ModelTransformer` expected that all logic about insertion FQ will be done inside, without call another backend-specific functions.

It is impossible to insert FQs without calling backend-specific functions.

> 
> > Otherwise you are breaking the priority functionality of insertions into the PT models.
> 
> Please provide example of incorrect work of PT PTQ.

You won't be able to apply PTQ as an extra step on top of a pruned or sparsified model, for instance.

In general - just why are you repeating the code that is already there in the NNCF PT code base to insert quantizers? Why aren't you reusing the existing functionality? If the existing code is not convenient to use for the PTQ right now - refactor it so that it becomes convenient both for PTQ and QAT.

Reply:
`ModelTransfomer` refactoring is already planned, ticket 102414.

Reply:
> I'm pretty sure you can reuse most of the code in nncf.torch.model_transformer.PTModelTransformer._apply_insertion_transformations by calling it inside your new _apply_quantizer_insertion_transformations.

Done